### ISSUE TITLE
[apex] Any class with an annotation starting with "@Invocable" must have a no-parameter constructor

### ISSUE THREAD

**Issue by dschach** [pmd/pmd#6647]
**Proposed Rule Name:**

Invocable Apex requires no-argument constructor

**Proposed Category:** One of [Best Practices | Code Style | Design | Documentation | Error Prone | Multithreading | Performance | Security]

Best Practices

**Description:**

> Apex classes used for invocable action parameters must have a visible no-argument constructor. The constructor must be public for non-packaged classes or global for packaged classes invoked from outside the package. Use API version 65.0 or earlier with the release update disabled to maintain the previous behavior. 

https://help.salesforce.com/s/articleView?id=release-notes.rn_apex_constructor_visibility_invocable_custom_classes_v66.htm&release=262&type=5

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

*BAD*
```apex
public class MyInvocableClass {
    @InvocableVariable(required=true)
    public String accountName;
    
    @InvocableVariable
    public Integer priority;
    
    // No explicit constructor - Apex provides an implicit public no-argument constructor
}
```

*GOOD*
```apex
public class MyInvocableClass {
    @InvocableVariable(required=true)
    public String accountName;
    
    @InvocableVariable
    public Integer priority;
    
    // Constructor with parameters
    public MyInvocableClass(String accountName, Integer priority) {
        this.accountName = accountName;
        this.priority = priority;
    }
    
    // Explicit no-argument constructor now required
    public MyInvocableClass() {
    }
}
```

**Possible Properties:**

*   Should this rule be customizable via properties?

I can't think of any
